### ISSUE TITLE
Show timer in topbar under Linux

### ISSUE THREAD

**Issue by sojusnik** [toggl/toggldesktop#2154]
Proposing to add a timer instead of the generic Toggl icon to the [topbar](http://i.imgur.com/csOQuRK.png) of Linux, f.i. on Ubuntu Gnome, like it's already implemented in OSX.

**Comment by IndrekV** [comment]
This need some investigation. I'm not sure this is easily doable. From the top of my head I can't name any applications on Linux that have text in their top bar/menubar icon.

**Comment by sojusnik** [comment]
It seems pretty straightforward to me. [This](https://extensions.gnome.org/extension/604/teatime/) Gnome extension has it implemented graphically and as a timer, as shown [here](http://i.imgur.com/9dCjjmm.png).

**Comment by IndrekV** [comment]
Awesome! I'll check it out.

**Comment by sojusnik** [comment]
Maybe you could also combine this with a [pomodoro timer](https://github.com/toggl/toggldesktop/issues/2017).

**Comment by SergiArias** [comment]
Yeah! Please, this would be awesome, as the desktop application is very unuseful compared to Windows & Mac.

**Comment by carlisgg** [comment]
+1 for this. That will be very useful and a game changer

**Comment by SergiArias** [comment]
I think it can be done by using [this guide](https://askubuntu.com/questions/259174/add-a-custom-text-to-gnome-panel) and the [show current time](https://github.com/toggl/toggl_api_docs/blob/master/chapters/time_entries.md#get-running-time-entry) entry method in the API. I will try it when I have some time and share my thoughts.

Please, if someone has some time and coding abilities to try, please do it as I will be slow :)

**Comment by keelbo** [comment]
Any update on this?

**Comment by MartinBriza** [comment]
There are no concrete plans on what to do about this because we'd have to implement a full GNOME (or Plasma, or Mate, or...) extension to do this. There are also other use cases for DE extensions that we could implement in the future...
The thing is, I'm not saying this cannot happen in the future BUT we're not planning to do this at this moment, sorry. I'll leave this issue open so we know there's demand for this feature.

**Comment by daroczig** [comment]
If anyone looking for this feature in Gnome 3, I've been using https://github.com/p-e-w/argos with a one-liner script to report on the current task and the timer. It's not interactive, so you cannot click there to switch task etc ... but it does update every few seconds, even added some color codes per project.

**Comment by MartinBriza** [comment]
I don't think we're implementing a GNOME shell extension any time soon but if anybody from the community wants to do so, I'm open to creating a simple DBus API for data retrieval. Feel free to ping me at martin@toggl.com for working out a specification for the API.

**Comment by ni-ka** [comment]
Stumbled over this as I would also like to display the current task on Gnome,

@MartinBriza I think a DBus API would be great. Maybe for inspiration check https://github.com/gnome-pomodoro/gnome-pomodoro which send DBus notifications on start / pause / time elapsed etc. See below for an example captured with dbus-monitor, BTW inbetween is TogglDesktop which updates its tooltip every 500ms even when the timer is not running, even though that tooltip is not visible on Gnome (maybe it is on KDE).

```
signal time=1614379412.486838 sender=:1.36 -> destination=(null destination) serial=400 path=/org/gnome/Pomodoro; interface=org.freedesktop.DBus.Properties; member=PropertiesChanged
   string "org.gnome.Pomodoro"
   array [
      dict entry(
         string "State"
         variant             string "pomodoro"
      )
      dict entry(
         string "Elapsed"
         variant             double 0
      )
      dict entry(
         string "StateDuration"
         variant             double 1500
      )
   ]
   array [
   ]
signal time=1614379412.749477 sender=:1.4637 -> destination=(null destination) serial=2754 path=/StatusNotifierItem; interface=org.kde.StatusNotifierItem; member=NewToolTip
signal time=1614379413.244812 sender=:1.4637 -> destination=(null destination) serial=2755 path=/StatusNotifierItem; interface=org.kde.StatusNotifierItem; member=NewToolTip
signal time=1614379413.374838 sender=:1.36 -> destination=(null destination) serial=401 path=/org/gnome/Pomodoro; interface=org.freedesktop.DBus.Properties; member=PropertiesChanged
   string "org.gnome.Pomodoro"
   array [
      dict entry(
         string "Elapsed"
         variant             double 1.0788
      )
   ]
   array [
   ]

```

@daroczig is your script available anywhere?

**Comment by alexandrebarbaruiva** [comment]
@MartinBriza I'd very much like to work on it, however I tried sending an email and received a response saying that your email doesn't exist...

**Comment by MartinBriza** [comment]
Yup, sadly I haven't been working for Toggl for over a year and as far as I know, there's no further development happening in the Linux app - or at least, that's what the project homepage implies.

**Comment by alexandrebarbaruiva** [comment]
Sad to hear it. Wish you success on your journey!
